src/gui/DatabaseOpenWidget.cpp

```diff
@@ -68,6 +69,16 @@ DatabaseOpenWidget::DatabaseOpenWidget(QWidget* parent)
         m_ui->editPassword->setShowPassword(false);
     });
 
+    m_fileExistsTimer.setInterval(fileExistsCheckInterval);
+    m_fileExistsTimer.setSingleShot(false);
+    connect(&m_fileExistsTimer, &QTimer::timeout, this, [this] {
+        if (!QFile::exists(m_filename)) {
+            m_ui->messageWidget->showMessage(tr("The database file does not exist or is not accessible."),
+                                             MessageWidget::Warning,
+                                             fileExistsCheckInterval + 500);
+        }
+    });
```

Comment:
Not sure whether I like this. It's a bit confusing if such a message suddenly pops up after an arbitrary time interval. Also, what happens in between when the user tries to unlock the non-existent file?
This also causes an unnecessary check after 5sec for every database. At least the timer should be stopped when the database is loaded correctly.

Reply:
A file can become unavailable at any time after being successfully detected/loaded. If network share gets severed, etc. 

Perhaps that check should be moved to the open sequence and refuse to open if the file cannot be read. Technically that already happens but deep in the database code with direct file access.